### PR TITLE
Bad WriterInterface implementation

### PR DESCRIPTION
**Pull request by naprstek**
AbstractWritter::setFormatter($formatter, array $options = null) does not conform WriterInterface::setFormatter($formatter).

Changed WriterInterface and all descendat subclasses.

Tests has the same result as master.

### CODE REVIEW

**Comment by michalbundyra** [comment]
@naprstek Duplicate of #79, but here I can see more changes

**Comment by Xerkus** [comment]
This is a BC break due to how php 7.2 handles addition of optional parameters to interfaces and overriden parents methods.

**Comment by samsonasik** [comment]
@naprstek can you register php 7.2 to .travis.yml file ?

**Comment by naprstek** [comment]
file updated (hope it is correct).

V.


-----
S pozdravem
Vladimír Náprstek

2018-04-09 19:14 GMT+02:00 Abdul Malik Ikhsan <notifications@github.com>:

**Comment by samsonasik** [comment]
the `zend-validator` requirement at composer.json seems need to be upped to version at least `^2.10.1` which fixes the `idn_to_utf8()` error at php 7.2 environment. Once it updated, `composer.lock` need to be updated by run : `composer update --prefer-lowest --prefer-stable` and commit it. /cc @webimpress  if any better suggestion

**Comment by samsonasik** [comment]
needs rebase

File: src/Writer/WriterInterface.php [hunk 1]
```diff
@@ -26,9 +26,10 @@ public function addFilter($filter);
      * Set a message formatter for the writer
      *
      * @param string|Formatter $formatter
+     * @param  array|null $options
      * @return WriterInterface
      */
-    public function setFormatter($formatter);
+    public function setFormatter($formatter, array $options = null);
```
Comment (weierophinney) [inline src/Writer/WriterInterface.php]: Please remove this change. It's a BC break (all implementations would need to be updated, including _userland_ implementations).

The reason PHP 7.2 was raising issues was due to how the `AbstractWriter` is defined, as it adds the additional, optional argument. Any class that extends _that_ class needs to be updated to comply with PHP 7.2, but the interface can remain as-is.

**Review by weierophinney — CHANGES_REQUESTED** [review]

**Comment by weierophinney** [comment]
@naprstek I've rebased for you, and incorporated the changes I suggested.

**Comment by weierophinney** [comment]
THanks, @naprstek!

**Pull request merged**